### PR TITLE
0.12.0 api files

### PR DESCRIPTION
**Pull request by nomisRev**
Diff between Arrow's 0.11.0 & 0.12.0 API.

### CODE REVIEW

File: arrow-core-data.api [hunk 1]
```diff
@@ -822,8 +914,12 @@ public final class arrow/core/Ior$Companion {
 	public final fun bothNel (Ljava/lang/Object;Ljava/lang/Object;)Larrow/core/Ior;
 	public final fun fromNullables (Ljava/lang/Object;Ljava/lang/Object;)Larrow/core/Ior;
 	public final fun fromOptions (Larrow/core/Option;Larrow/core/Option;)Larrow/core/Option;
+	public final fun getUnit ()Larrow/core/Ior;
```
Comment (nomisRev) [inline arrow-core-data.api]: Revert?

File: arrow-core-data.api [hunk 2]
```diff
@@ -822,8 +914,12 @@ public final class arrow/core/Ior$Companion {
...
 	public final fun leftNel (Ljava/lang/Object;)Larrow/core/Ior;
+	public final fun lift (Lkotlin/jvm/functions/Function1;)Lkotlin/jvm/functions/Function1;
+	public final fun lift (Lkotlin/jvm/functions/Function1;Lkotlin/jvm/functions/Function1;)Lkotlin/jvm/functions/Function1;
 	public final fun tailRecM (Ljava/lang/Object;Lkotlin/jvm/functions/Function1;Larrow/typeclasses/Semigroup;)Larrow/core/Ior;
+	public final fun unit ()Larrow/core/Ior;
```
Comment (nomisRev) [inline arrow-core-data.api]: Revert?

File: arrow-core-data.api [hunk 3]
```diff
@@ -797,9 +882,16 @@ public abstract class arrow/core/Ior : arrow/Kind {
...
 	public final fun toLeftOption ()Larrow/core/Option;
 	public final fun toOption ()Larrow/core/Option;
+	public fun toString ()Ljava/lang/String;
 	public final fun toValidated ()Larrow/core/Validated;
 	public final fun traverse (Larrow/typeclasses/Applicative;Lkotlin/jvm/functions/Function1;)Larrow/Kind;
+	public final fun traverse (Lkotlin/jvm/functions/Function1;)Ljava/util/List;
+	public final fun traverseEither (Lkotlin/jvm/functions/Function1;)Larrow/core/Either;
+	public final fun traverseValidated (Lkotlin/jvm/functions/Function1;)Larrow/core/Validated;
+	public final fun tupleLeft (Ljava/lang/Object;)Larrow/core/Ior;
+	public final fun tupleRight (Ljava/lang/Object;)Larrow/core/Ior;
```
Comment (nomisRev) [inline arrow-core-data.api]: Do we want this for all data types?

File: arrow-core-data.api [hunk 4]
```diff
@@ -747,47 +838,41 @@ public final class arrow/core/Hashed$Companion {
...
 	public final fun foldRight (Larrow/core/Eval;Lkotlin/jvm/functions/Function2;)Larrow/core/Eval;
+	public final fun fproduct (Lkotlin/jvm/functions/Function1;)Larrow/core/Ior;
 	public abstract fun isBoth ()Z
+	public final fun isEmpty ()Z
 	public abstract fun isLeft ()Z
+	public final fun isNotEmpty ()Z
 	public abstract fun isRight ()Z
 	public final fun leftOrNull ()Ljava/lang/Object;
 	public final fun map (Lkotlin/jvm/functions/Function1;)Larrow/core/Ior;
+	public final fun mapConst (Ljava/lang/Object;)Larrow/core/Ior;
```
Comment (nomisRev) [inline arrow-core-data.api]: Do we want this simply alias over `map { _ -> a }`

File: arrow-core-data.api [hunk 5]
```diff
@@ -866,13 +962,107 @@ public final class arrow/core/Ior$Right$Companion {
 
 public final class arrow/core/IorKt {
 	public static final fun ap (Larrow/core/Ior;Larrow/typeclasses/Semigroup;Larrow/Kind;)Larrow/core/Ior;
+	public static final fun apEval (Larrow/core/Ior;Larrow/typeclasses/Semigroup;Larrow/core/Eval;)Larrow/core/Eval;
```
Comment (nomisRev) [inline arrow-core-data.api:775]: `ap` is questionable, so we should definitely question `apEval`.

File: arrow-core-data.api [hunk 6]
```diff
@@ -866,13 +962,107 @@ public final class arrow/core/Ior$Right$Companion {
...
 	public static final fun getOrElse (Larrow/core/Ior;Lkotlin/jvm/functions/Function0;)Ljava/lang/Object;
+	public static final fun ifM (Larrow/core/Ior;Larrow/typeclasses/Semigroup;Lkotlin/jvm/functions/Function0;Lkotlin/jvm/functions/Function0;)Larrow/core/Ior;
+	public static final fun ior (Larrow/typeclasses/Semigroup$Companion;Larrow/typeclasses/Semigroup;Larrow/typeclasses/Semigroup;)Larrow/typeclasses/Semigroup;
 	public static final fun leftIor (Ljava/lang/Object;)Larrow/core/Ior;
+	public static final fun leftWiden (Larrow/core/Ior;)Larrow/core/Ior;
+	public static final fun mproduct (Larrow/core/Ior;Larrow/typeclasses/Semigroup;Lkotlin/jvm/functions/Function1;)Larrow/core/Ior;
+	public static final fun replicate (Larrow/core/Ior;Larrow/typeclasses/Semigroup;I)Larrow/core/Ior;
+	public static final fun replicate (Larrow/core/Ior;Larrow/typeclasses/Semigroup;ILarrow/typeclasses/Monoid;)Larrow/core/Ior;
 	public static final fun rightIor (Ljava/lang/Object;)Larrow/core/Ior;
+	public static final fun selectM (Larrow/core/Ior;Larrow/typeclasses/Semigroup;Larrow/core/Ior;)Larrow/core/Ior;
```
Comment (nomisRev) [inline arrow-core-data.api]: `selectM` part of `Selective` API?

File: arrow-core-data.api [hunk 7]
```diff
@@ -866,13 +962,107 @@ public final class arrow/core/Ior$Right$Companion {
...
+	public static final fun zip (Larrow/core/Ior;Larrow/typeclasses/Semigroup;Larrow/core/Ior;Larrow/core/Ior;Larrow/core/Ior;Larrow/core/Ior;Larrow/core/Ior;Lkotlin/jvm/functions/Function6;)Larrow/core/Ior;
+	public static final fun zip (Larrow/core/Ior;Larrow/typeclasses/Semigroup;Larrow/core/Ior;Larrow/core/Ior;Larrow/core/Ior;Larrow/core/Ior;Lkotlin/jvm/functions/Function5;)Larrow/core/Ior;
+	public static final fun zip (Larrow/core/Ior;Larrow/typeclasses/Semigroup;Larrow/core/Ior;Larrow/core/Ior;Larrow/core/Ior;Lkotlin/jvm/functions/Function4;)Larrow/core/Ior;
+	public static final fun zip (Larrow/core/Ior;Larrow/typeclasses/Semigroup;Larrow/core/Ior;Larrow/core/Ior;Lkotlin/jvm/functions/Function3;)Larrow/core/Ior;
+	public static final fun zip (Larrow/core/Ior;Larrow/typeclasses/Semigroup;Larrow/core/Ior;Lkotlin/jvm/functions/Function2;)Larrow/core/Ior;
+	public static final fun zipEval (Larrow/core/Ior;Larrow/typeclasses/Semigroup;Larrow/core/Eval;Lkotlin/jvm/functions/Function2;)Larrow/core/Eval;
```
Comment (nomisRev) [inline arrow-core-data.api]: Idem as `apEval`. Not sure if adding this API is meaningful.

File: arrow-core-data.api [hunk 8]
```diff
@@ -866,13 +962,107 @@ public final class arrow/core/Ior$Right$Companion {
...
+	public static final fun crosswalkMap (Ljava/lang/Iterable;Lkotlin/jvm/functions/Function1;)Ljava/util/Map;
+	public static final fun crosswalkNull (Ljava/lang/Iterable;Lkotlin/jvm/functions/Function1;)Ljava/util/List;
+	public static final fun flatten (Ljava/lang/Iterable;)Ljava/util/List;
+	public static final fun fold (Ljava/lang/Iterable;Larrow/typeclasses/Monoid;)Ljava/lang/Object;
+	public static final fun foldMap (Ljava/lang/Iterable;Larrow/typeclasses/Monoid;Lkotlin/jvm/functions/Function1;)Ljava/lang/Object;
+	public static final fun foldRight (Ljava/lang/Iterable;Ljava/lang/Object;Lkotlin/jvm/functions/Function2;)Ljava/lang/Object;
+	public static final fun foldRight (Ljava/util/List;Larrow/core/Eval;Lkotlin/jvm/functions/Function2;)Larrow/core/Eval;
+	public static final fun fproduct (Ljava/lang/Iterable;Lkotlin/jvm/functions/Function1;)Ljava/util/List;
+	public static final fun ifM (Ljava/lang/Iterable;Lkotlin/jvm/functions/Function0;Lkotlin/jvm/functions/Function0;)Ljava/util/List;
+	public static final fun ifThen (Ljava/lang/Iterable;Ljava/lang/Iterable;Lkotlin/jvm/functions/Function1;)Ljava/lang/Iterable;
```
Comment (nomisRev) [inline arrow-core-data.api]: Do we want these? They come back in other data types, but seem to be new in most datatypes.

File: arrow-core-data.api [hunk 9]
```diff
@@ -2150,32 +3297,56 @@ public final class arrow/core/UtilsKt {
...
-	public final fun catchNel (Lkotlin/jvm/functions/Function1;Lkotlin/coroutines/Continuation;)Ljava/lang/Object;
+	public final fun catch (Lkotlin/jvm/functions/Function0;)Larrow/core/Validated;
+	public final synthetic fun catch (Lkotlin/jvm/functions/Function1;Lkotlin/coroutines/Continuation;)Ljava/lang/Object;
+	public final fun catch (Lkotlin/jvm/functions/Function1;Lkotlin/jvm/functions/Function0;)Larrow/core/Validated;
+	public final fun catchNel (Lkotlin/jvm/functions/Function0;)Larrow/core/Validated;
+	public final synthetic fun catchNel (Lkotlin/jvm/functions/Function1;Lkotlin/coroutines/Continuation;)Ljava/lang/Object;
 	public final fun fromEither (Larrow/core/Either;)Larrow/core/Validated;
 	public final fun fromNullable (Ljava/lang/Object;Lkotlin/jvm/functions/Function0;)Larrow/core/Validated;
 	public final fun fromOption (Larrow/core/Option;Lkotlin/jvm/functions/Function0;)Larrow/core/Validated;
+	public final fun getS ()I
```
Comment (nomisRev) [inline arrow-core-data.api]: ?

File: arrow-core.api [hunk 10]
```diff
@@ -973,6 +967,7 @@ public final class arrow/core/extensions/ConstTraverseFilter$DefaultImpls {
 	public static fun foldLeft (Larrow/core/extensions/ConstTraverseFilter;Larrow/Kind;Ljava/lang/Object;Lkotlin/jvm/functions/Function2;)Ljava/lang/Object;
 	public static fun foldM (Larrow/core/extensions/ConstTraverseFilter;Larrow/Kind;Larrow/typeclasses/Monad;Ljava/lang/Object;Lkotlin/jvm/functions/Function2;)Larrow/Kind;
 	public static fun foldMap (Larrow/core/extensions/ConstTraverseFilter;Larrow/Kind;Larrow/typeclasses/Monoid;Lkotlin/jvm/functions/Function1;)Ljava/lang/Object;
+	public static fun foldMapA (Larrow/core/extensions/ConstTraverseFilter;Larrow/Kind;Larrow/typeclasses/Applicative;Larrow/typeclasses/Monoid;Lkotlin/jvm/functions/Function1;)Larrow/Kind;
```
Comment (nomisRev) [inline arrow-core.api]: `foldMapA` seems to be added ???? Noticed this change too in `arrow-core-data`, is this a new method added in a typeclass?

File: arrow-core.api [hunk 11]
```diff
@@ -16209,6 +13830,11 @@ public final class arrow/core/extensions/set/semigroupal/SetKSemigroupalKt {
 	public static final fun times (Ljava/util/Set;Ljava/util/Set;)Ljava/util/Set;
 }
 
+public final class arrow/core/extensions/set/zip/SetKZipKt {
+	public static final fun zip (Ljava/util/Set;Ljava/util/Set;)Ljava/util/Set;
+	public static final fun zipWith (Ljava/util/Set;Ljava/util/Set;Lkotlin/jvm/functions/Function2;)Ljava/util/Set;
+}
+
```
Comment (nomisRev) [inline arrow-core.api]: Remove.

File: arrow-fx-coroutines.api [hunk 12]
```diff
@@ -103,24 +104,31 @@ public final class arrow/fx/coroutines/CancelToken$Companion {
 	public final fun getUnit-CF6U7qU ()Lkotlin/jvm/functions/Function1;
 }
 
-public abstract interface class arrow/fx/coroutines/CancellableContinuation : kotlin/coroutines/Continuation {
+public abstract class arrow/fx/coroutines/CancellableContinuation : kotlin/coroutines/Continuation {
```
Comment (nomisRev) [inline arrow-fx-coroutines.api]: Revert change. This is deprecated anyway.

File: arrow-fx-coroutines.api [hunk 13]
```diff
@@ -506,22 +619,56 @@ public abstract class arrow/fx/coroutines/Resource {
...
+	public fun <init> (Larrow/fx/coroutines/Resource;Lkotlin/jvm/functions/Function1;)V
+	public final fun getF ()Lkotlin/jvm/functions/Function1;
+	public final fun getSource ()Larrow/fx/coroutines/Resource;
 }
 
 public final class arrow/fx/coroutines/Resource$Companion {
 	public final fun defer (Lkotlin/jvm/functions/Function1;)Larrow/fx/coroutines/Resource;
+	public final fun fromClosable (Lkotlin/jvm/functions/Function1;)Larrow/fx/coroutines/Resource;
 	public final fun fromExecutor (Lkotlin/jvm/functions/Function1;)Larrow/fx/coroutines/Resource;
+	public final fun getUnit ()Larrow/fx/coroutines/Resource;
```
Comment (nomisRev) [inline arrow-fx-coroutines.api]: Revert. We're not exposing `unit` constructors anymore.

File: arrow-fx-coroutines.api [hunk 14]
```diff
@@ -506,22 +619,56 @@ public abstract class arrow/fx/coroutines/Resource {
...
+	public fun <init> (Lkotlin/jvm/functions/Function1;)V
+	public final fun getResource ()Lkotlin/jvm/functions/Function1;
+}
+
 public final class arrow/fx/coroutines/ResourceKt {
-	public static final fun release-2-trE80 (Lkotlin/jvm/functions/Function1;Lkotlin/jvm/functions/Function2;)Larrow/fx/coroutines/Resource;
-	public static final fun releaseCase-JxnonM0 (Lkotlin/jvm/functions/Function1;Lkotlin/jvm/functions/Function3;)Larrow/fx/coroutines/Resource;
+	public static final fun flatSequence (Ljava/lang/Iterable;)Larrow/fx/coroutines/Resource;
+	public static final fun flatTraverseFilterResource (Ljava/lang/Iterable;Lkotlin/jvm/functions/Function1;)Larrow/fx/coroutines/Resource;
+	public static final fun flatTraverseResource (Ljava/lang/Iterable;Lkotlin/jvm/functions/Function1;)Larrow/fx/coroutines/Resource;
```
Comment (nomisRev) [inline arrow-fx-coroutines.api]: `flat` variants need to be removed. We're not adding these at this point.

File: arrow-fx-coroutines.api [hunk 15]
```diff
@@ -506,22 +619,56 @@ public abstract class arrow/fx/coroutines/Resource {
...
+	public static final fun release-zgiIeyo (Lkotlin/jvm/functions/Function1;Lkotlin/jvm/functions/Function2;)Larrow/fx/coroutines/Resource;
+	public static final fun releaseCase-zgiIeyo (Lkotlin/jvm/functions/Function1;Lkotlin/jvm/functions/Function3;)Larrow/fx/coroutines/Resource;
 	public static final fun resource (Lkotlin/jvm/functions/Function1;)Lkotlin/jvm/functions/Function1;
+	public static final fun sequence (Ljava/lang/Iterable;)Larrow/fx/coroutines/Resource;
+	public static final fun traverseFilterResource (Ljava/lang/Iterable;Lkotlin/jvm/functions/Function1;)Larrow/fx/coroutines/Resource;
+	public static final fun traverseResource (Ljava/lang/Iterable;Lkotlin/jvm/functions/Function1;)Larrow/fx/coroutines/Resource;
```
Comment (nomisRev) [inline arrow-fx-coroutines.api]: `filter` variants need to be removed. We're not adding these at this point.

File: arrow-fx-coroutines.api [hunk 16]
```diff
@@ -649,60 +840,25 @@ public final class arrow/fx/coroutines/Semaphore$DefaultImpls {
...
-	public fun toString ()Ljava/lang/String;
-	public fun tryReactivate ()Z
-}
-
 public final class arrow/fx/coroutines/TimerKt {
 	public static final fun sleep (Larrow/fx/coroutines/Duration;Lkotlin/coroutines/Continuation;)Ljava/lang/Object;
 	public static final fun timeOutOrNull (Larrow/fx/coroutines/Duration;Lkotlin/jvm/functions/Function1;Lkotlin/coroutines/Continuation;)Ljava/lang/Object;
 }
 
+public final class arrow/fx/coroutines/UnsafePromise {
```
Comment (nomisRev) [inline arrow-fx-coroutines.api:848]: Missing deprecation on master

File: arrow-fx-stm.api [hunk 17]
```diff
@@ -0,0 +1,314 @@
+public abstract interface class arrow/fx/stm/STM {
```
Comment (nomisRev) [inline arrow-fx-stm.api:1]: Hard remove the current only deprecation. Since this library is newly added it can go straight to 0.13.0 API.

**Review by nomisRev — COMMENTED** [review]
Initial review.

File: arrow-continuations.api [hunk 18]
```diff
@@ -1,123 +1,45 @@
...
-	public fun <init> (Lkotlin/coroutines/Continuation;Ljava/util/List;)V
-	public final fun copy (Lkotlin/coroutines/Continuation;Ljava/util/List;)Larrow/continuations/generic/DelimContScope$SingleShotCont;
-	public static synthetic fun copy$default (Larrow/continuations/generic/DelimContScope$SingleShotCont;Lkotlin/coroutines/Continuation;Ljava/util/List;ILjava/lang/Object;)Larrow/continuations/generic/DelimContScope$SingleShotCont;
-	public fun equals (Ljava/lang/Object;)Z
-	public fun hashCode ()I
-	public fun invoke (Ljava/lang/Object;Lkotlin/coroutines/Continuation;)Ljava/lang/Object;
-	public fun toString ()Ljava/lang/String;
+public class arrow/continuations/generic/ControlThrowable : java/lang/Throwable {
+	public fun <init> ()V
+	public fun fillInStackTrace ()Ljava/lang/Throwable;
```
Comment (raulraja) [inline arrow-continuations.api:19]: Wondering if this stacktrace is actually useful in the context of structured concurrency if someone has a massive `either` block nested

File: arrow-core.api [hunk 19]
```diff
@@ -5883,6 +4109,7 @@ public final class arrow/core/extensions/ListKTraverse$DefaultImpls {
 	public static fun foldLeft (Larrow/core/extensions/ListKTraverse;Larrow/Kind;Ljava/lang/Object;Lkotlin/jvm/functions/Function2;)Ljava/lang/Object;
 	public static fun foldM (Larrow/core/extensions/ListKTraverse;Larrow/Kind;Larrow/typeclasses/Monad;Ljava/lang/Object;Lkotlin/jvm/functions/Function2;)Larrow/Kind;
 	public static fun foldMap (Larrow/core/extensions/ListKTraverse;Larrow/Kind;Larrow/typeclasses/Monoid;Lkotlin/jvm/functions/Function1;)Ljava/lang/Object;
+	public static fun foldMapA (Larrow/core/extensions/ListKTraverse;Larrow/Kind;Larrow/typeclasses/Applicative;Larrow/typeclasses/Monoid;Lkotlin/jvm/functions/Function1;)Larrow/Kind;
```
Comment (raulraja) [inline arrow-core.api]: to discuss in review the foldMapX variants and cryptic names like foldM

File: arrow-fx-coroutines-test.api [hunk 20]
```diff
@@ -0,0 +1,105 @@
...
+	public static final fun nullable (Lio/kotest/property/Arb$Companion;Lio/kotest/property/Arb;)Lio/kotest/property/Arb;
+	public static final fun rethrow (Larrow/core/Either;)Ljava/lang/Object;
+	public static final fun shift (Lkotlin/coroutines/CoroutineContext;Lkotlin/coroutines/Continuation;)Ljava/lang/Object;
+	public static final fun suspend (Ljava/lang/Object;Lkotlin/coroutines/Continuation;)Ljava/lang/Object;
+	public static final fun suspend (Ljava/lang/Throwable;Lkotlin/coroutines/Continuation;)Ljava/lang/Object;
+	public static final fun suspended (Ljava/lang/Object;)Lkotlin/jvm/functions/Function1;
+	public static final fun throwable (Lio/kotest/property/Arb$Companion;)Lio/kotest/property/Arb;
+	public static final fun toEither (Ljava/lang/Object;)Larrow/core/Either;
+	public static final fun unit (Lio/kotest/property/Arb$Companion;)Lio/kotest/property/Arb;
+	public static final fun unsafeEquals-MiKvLTk (Lkotlin/jvm/functions/Function1;)Lio/kotest/matchers/Matcher;
```
Comment (raulraja) [inline arrow-fx-coroutines-test.api:49]: LOL what is `MiKvLTk`?

File: arrow-fx-coroutines.api [hunk 21]
```diff
@@ -279,7 +304,7 @@ public abstract interface class arrow/fx/coroutines/Fiber {
 }
 
 public final class arrow/fx/coroutines/Fiber$Companion {
-	public final fun invoke-EdtQ1QA (Lkotlin/jvm/functions/Function1;Lkotlin/jvm/functions/Function1;)Larrow/fx/coroutines/Fiber;
+	public final fun invoke-9c1J1T8 (Lkotlin/jvm/functions/Function1;Lkotlin/jvm/functions/Function1;)Larrow/fx/coroutines/Fiber;
```
Comment (raulraja) [inline arrow-fx-coroutines.api:307]: does this generate a new id each time it compiles?

File: arrow-fx-coroutines.api [hunk 22]
```diff
@@ -506,22 +619,56 @@ public abstract class arrow/fx/coroutines/Resource {
...
 public final class arrow/fx/coroutines/Resource$Companion {
 	public final fun defer (Lkotlin/jvm/functions/Function1;)Larrow/fx/coroutines/Resource;
+	public final fun fromClosable (Lkotlin/jvm/functions/Function1;)Larrow/fx/coroutines/Resource;
 	public final fun fromExecutor (Lkotlin/jvm/functions/Function1;)Larrow/fx/coroutines/Resource;
+	public final fun getUnit ()Larrow/fx/coroutines/Resource;
 	public final fun invoke (Lkotlin/jvm/functions/Function1;Lkotlin/jvm/functions/Function2;)Larrow/fx/coroutines/Resource;
 	public final fun invoke (Lkotlin/jvm/functions/Function1;Lkotlin/jvm/functions/Function3;)Larrow/fx/coroutines/Resource;
 	public final fun just (Ljava/lang/Object;)Larrow/fx/coroutines/Resource;
 	public final fun singleThreadContext (Ljava/lang/String;)Larrow/fx/coroutines/Resource;
 	public final fun tailRecM (Ljava/lang/Object;Lkotlin/jvm/functions/Function1;)Larrow/fx/coroutines/Resource;
```
Comment (raulraja) [inline arrow-fx-coroutines.api:653]: if we are keeping any of the `tailRecM` methods I think we need to change their name to something more friendly perhaps simply `tailRecurse` or similar now the M word is out of the pic for the most part

File: arrow-fx-stm.api (hunk 17 above)
Comment (raulraja) [reply arrow-fx-stm.api:1]: We should perhaps review the type names in this module for 0.13

File: arrow-optics.api [hunk 23]
```diff
@@ -229,32 +201,69 @@ public final class arrow/optics/Getter$DefaultImpls {
 	public static fun zip (Larrow/optics/Getter;Larrow/optics/Getter;)Larrow/optics/Getter;
 }
 
-public final class arrow/optics/IdKt {
-	public static final fun toPValue (Larrow/core/Id$Companion;)Larrow/optics/PIso;
-	public static final fun toValue (Larrow/core/Id$Companion;)Larrow/optics/PIso;
+public final class arrow/optics/GetterKt {
```
Comment (raulraja) [inline arrow-optics.api:204]: Curious with the new subtyping encoding how the java generation part looks like and if it's friendlier to consume from Javawithout the _Kt files

File: arrow-syntax.api [hunk 24]
```diff
@@ -0,0 +1,1632 @@
...
+	public final fun getF ()Lkotlin/jvm/functions/Function0;
+	public final fun getIndex ()I
+	public fun hashCode ()I
+	public fun toString ()Ljava/lang/String;
+}
+
+public abstract class arrow/core/AndThen1 : kotlin/jvm/functions/Function1 {
+	public static final field Companion Larrow/core/AndThen1$Companion;
+	public final fun andThen (Lkotlin/jvm/functions/Function1;)Larrow/core/AndThen1;
+	public final fun andThenF (Larrow/core/AndThen1;)Larrow/core/AndThen1;
```
Comment (raulraja) [inline arrow-syntax.api]: We should look into these names with `F` and similar

File: arrow-syntax.api [hunk 25]
```diff
@@ -0,0 +1,1632 @@
...
+
+public final class arrow/core/PairingKt {
+	public static final fun paired (Lkotlin/jvm/functions/Function2;)Lkotlin/jvm/functions/Function1;
+	public static final fun tripled (Lkotlin/jvm/functions/Function3;)Lkotlin/jvm/functions/Function1;
+	public static final fun unpaired (Lkotlin/jvm/functions/Function1;)Lkotlin/jvm/functions/Function2;
+	public static final fun untripled (Lkotlin/jvm/functions/Function1;)Lkotlin/jvm/functions/Function3;
+}
+
+public final class arrow/core/PartialsKt {
+	public static final fun partially1 (Lkotlin/jvm/functions/Function10;Ljava/lang/Object;)Lkotlin/jvm/functions/Function9;
```
Comment (raulraja) [inline arrow-syntax.api]: do these need the number appended or would resolution work without it by using the lambda arg arity?

File: arrow-syntax.api [hunk 26]
```diff
@@ -0,0 +1,1632 @@
...
+	public static final fun plus (Larrow/core/Tuple3;Ljava/lang/Object;)Larrow/core/Tuple4;
+	public static final fun plus (Larrow/core/Tuple4;Ljava/lang/Object;)Larrow/core/Tuple5;
+	public static final fun plus (Larrow/core/Tuple5;Ljava/lang/Object;)Larrow/core/Tuple6;
+	public static final fun plus (Larrow/core/Tuple6;Ljava/lang/Object;)Larrow/core/Tuple7;
+	public static final fun plus (Larrow/core/Tuple7;Ljava/lang/Object;)Larrow/core/Tuple8;
+	public static final fun plus (Larrow/core/Tuple8;Ljava/lang/Object;)Larrow/core/Tuple9;
+	public static final fun plus (Larrow/core/Tuple9;Ljava/lang/Object;)Larrow/core/Tuple10;
+}
+
+public final class arrow/syntax/function/ComplementKt {
```
Comment (raulraja) [inline arrow-syntax.api:111]: I believe these are not useful

File: arrow-syntax.api [hunk 27]
```diff
@@ -0,0 +1,1632 @@
...
+	public static synthetic fun invoke$default (Lkotlin/jvm/functions/Function16;Larrow/syntax/function/Partial;Larrow/syntax/function/Partial;Larrow/syntax/function/Partial;Larrow/syntax/function/Partial;Larrow/syntax/function/Partial;Larrow/syntax/function/Partial;Larrow/syntax/function/Partial;Ljava/lang/Object;Larrow/syntax/function/Partial;Larrow/syntax/function/Partial;Larrow/syntax/function/Partial;Larrow/syntax/function/Partial;Larrow/syntax/function/Partial;Larrow/syntax/function/Partial;Larrow/syntax/function/Partial;Larrow/syntax/function/Partial;ILjava/lang/Object;)Lkotlin/jvm/functions/Function15;
+	public static synthetic fun invoke$default (Lkotlin/jvm/functions/Function16;Larrow/syntax/function/Partial;Larrow/syntax/function/Partial;Larrow/syntax/function/Partial;Larrow/syntax/function/Partial;Larrow/syntax/function/Partial;Larrow/syntax/function/Partial;Ljava/l
```
Comment (raulraja) [inline arrow-syntax.api:738]: I think this is something we can perhaps cut to more than half and not support so many arities as arriving at this place is extremely obscure and rare.

Comment (raulraja) [inline arrow-syntax.api:1064]: I believe this is not useful either in most FP scenarios in Kotlin

Comment (raulraja) [inline arrow-syntax.api:1039]: Not sure if useful

File: arrow-continuations.api (hunk 18 above)
Comment (nomisRev) [reply arrow-continuations.api:19]: Note: we can introduce `arrow.fillstacktrace=enabled` to generate stacktraces for all currently no-op stacktraces later.

File: arrow-fx-coroutines.api (hunk 22 above)
Comment (nomisRev) [reply arrow-fx-coroutines.api:653]: `tailRecM` needs to be deprecated.

File: arrow-syntax.api (hunk 24 above)
Comment (nomisRev) [reply arrow-syntax.api]: Make these things `private`.

File: arrow-syntax.api (hunk 25 above)
Comment (nomisRev) [reply arrow-syntax.api]: Reduce to arity-5

File: arrow-syntax.api (hunk 26 above)
Comment (nomisRev) [reply arrow-syntax.api:111]: Remove

File: arrow-syntax.api (hunk 27 above)
Comment (nomisRev) [reply arrow-syntax.api:738]: Already deprecated

Comment (nomisRev) [reply arrow-syntax.api:1064]: Already deprecated

Comment (nomisRev) [reply arrow-syntax.api:1039]: Already deprecated

File: arrow-syntax.api [hunk 28]
```diff
@@ -0,0 +1,1632 @@
...
+	public static final fun partially9 (Lkotlin/jvm/functions/Function17;Ljava/lang/Object;)Lkotlin/jvm/functions/Function16;
+	public static final fun partially9 (Lkotlin/jvm/functions/Function18;Ljava/lang/Object;)Lkotlin/jvm/functions/Function17;
+	public static final fun partially9 (Lkotlin/jvm/functions/Function19;Ljava/lang/Object;)Lkotlin/jvm/functions/Function18;
+	public static final fun partially9 (Lkotlin/jvm/functions/Function20;Ljava/lang/Object;)Lkotlin/jvm/functions/Function19;
+	public static final fun partially9 (Lkotlin/jvm/functions/Function21;Ljava/lang/Object;)Lkotlin/jvm/functions/Function20;
+	public static final fun partially9 (Lkotlin/jvm/functions/Function22;Ljava/lang/Object;)Lkotlin/jvm/functions/Function21;
+	public static final fun partially9 (Lkotlin/jvm/functions/Function9;Ljava/lang/Object;)Lkotlin/jvm/functions/Function8;
+}
+
+public final class arrow/core/PipeKt {
```
Comment (nomisRev) [inline arrow-syntax.api]: Deprecate in favor of normal function applications

File: arrow-syntax.api [hunk 29]
```diff
@@ -0,0 +1,1632 @@
...
+	public static final fun pipe4 (Ljava/lang/Object;Lkotlin/jvm/functions/Function4;)Lkotlin/jvm/functions/Function3;
+	public static final fun pipe5 (Ljava/lang/Object;Lkotlin/jvm/functions/Function5;)Lkotlin/jvm/functions/Function4;
+	public static final fun pipe6 (Ljava/lang/Object;Lkotlin/jvm/functions/Function6;)Lkotlin/jvm/functions/Function5;
+	public static final fun pipe7 (Ljava/lang/Object;Lkotlin/jvm/functions/Function7;)Lkotlin/jvm/functions/Function6;
+	public static final fun pipe8 (Ljava/lang/Object;Lkotlin/jvm/functions/Function8;)Lkotlin/jvm/functions/Function7;
+	public static final fun pipe9 (Ljava/lang/Object;Lkotlin/jvm/functions/Function9;)Lkotlin/jvm/functions/Function8;
+	public static final fun pipeLazy (Ljava/lang/Object;Lkotlin/jvm/functions/Function1;)Lkotlin/jvm/functions/Function0;
+}
+
+public final class arrow/core/ReverseKt {
```
Comment (nomisRev) [inline arrow-syntax.api]: Deprecate this as well.

File: arrow-syntax.api [hunk 30]
```diff
@@ -0,0 +1,1632 @@
...
+	public static final fun plus (Larrow/core/Tuple6;Ljava/lang/Object;)Larrow/core/Tuple7;
+	public static final fun plus (Larrow/core/Tuple7;Ljava/lang/Object;)Larrow/core/Tuple8;
+	public static final fun plus (Larrow/core/Tuple8;Ljava/lang/Object;)Larrow/core/Tuple9;
+	public static final fun plus (Larrow/core/Tuple9;Ljava/lang/Object;)Larrow/core/Tuple10;
+	public static final fun plus (Lkotlin/Pair;Ljava/lang/Object;)Lkotlin/Triple;
+	public static final fun plus (Lkotlin/Triple;Ljava/lang/Object;)Larrow/core/Tuple4;
+}
+
+public final class arrow/core/TuplingKt {
+	public static final fun tupled (Lkotlin/jvm/functions/Function10;)Lkotlin/jvm/functions/Function1;
```
Comment (nomisRev) [inline arrow-syntax.api]: Deprecate. Not useful since arity-n functions, and tupling is not supported in Kotlin.

File: arrow-syntax.api [hunk 31]
```diff
@@ -0,0 +1,1632 @@
...
+	public static final fun memoize (Lkotlin/jvm/functions/Function7;)Lkotlin/jvm/functions/Function7;
+	public static final fun memoize (Lkotlin/jvm/functions/Function8;)Lkotlin/jvm/functions/Function8;
+	public static final fun memoize (Lkotlin/jvm/functions/Function9;)Lkotlin/jvm/functions/Function9;
+}
+
+public final class arrow/syntax/function/OptionKt {
+	public static final fun optionLift (Lkotlin/jvm/functions/Function1;)Lkotlin/jvm/functions/Function1;
+}
+
+public final class arrow/syntax/function/PairingKt {
```
Comment (nomisRev) [inline arrow-syntax.api:266]: Deprecate. Not useful since arity-n functions, and tupling is not supported in Kotlin.

**Comment by nomisRev** [comment]
Closed due to 0.12.x & 0.13.x released

**Pull request closed without merging**